Index: les2_positioning.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- les2_positioning.py	(date 1550123816000)
+++ les2_positioning.py	(date 1550123816000)
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# @Time    : 2019-02-14 10:40
+# @Author  : kuxui
+# @Site    : 
+# @File    : les2_positioning.py
+# @Software: PyCharm
+# 功能：用多种方式定位京东登录页元素：id定位、nane定位、class name定位、tag name定位、linke text定位、
+#                              partial link text定位、xpath定位、css定位
+
+from selenium import webdriver
+from time import sleep
+
+
+# 清空用户名&密码输入框
+def clear():
+    driver.find_element_by_id("loginname").clear()
+    driver.find_element_by_id("nloginpwd").clear()
+
+
+# 1.启动浏览器
+driver = webdriver.Chrome()
+# 2.打开京东网页
+driver.get("https://passport.jd.com/new/login.aspx?ReturnUrl=https%3A%2F%2Fwww.jd.com%2F")
+# 3.扫码登录和账户登录，选择账户登录
+driver.find_element_by_link_text("账户登录").click()
+
+# HTML元素定位，定位用户名并输入值，定位密码并输入值
+# 1>id定位
+# sleep(2)
+# driver.find_element_by_id("loginname").send_keys("loginname")
+# sleep(2)
+# driver.find_element_by_id("nloginpwd").send_keys("nloginpwd")
+# sleep(2)
+# # 2>nane定位
+# clear()
+# sleep(2)
+# driver.find_element_by_name("loginname").send_keys("loginname")
+# sleep(2)
+# driver.find_element_by_name("nloginpwd").send_keys("nloginpwd")
+# sleep(2)
+# # 3>class name定位
+# clear()
+# sleep(2)
+# driver.find_element_by_class_name("itxt").send_keys("itxt")
+# sleep(2)
+# driver.find_element_by_class_name("itxt-error").send_keys("itxt-error")
+# sleep(2)
+# 4>tag name定位
+# clear()
+# sleep(2)
+# driver.find_elements_by_tag_name("input")[10].send_keys("10")
+# sleep(2)
+# driver.find_elements_by_tag_name("input")[11].send_keys("10")
+# sleep(2)
+# 5>linke text定位"忘记密码"
+# clear()
+# sleep(2)
+# driver.find_element_by_link_text("忘记密码").click()
+# 6>partial link text定位   截取文本其中一部分定位
+# clear()
+# sleep(2)
+# driver.find_element_by_partial_link_text("忘记").click()
+# sleep(2)
+# 7>xpath定位  绝对路径定位、相对路径定位、绝对路径加元素属性定位（属性的值一定要写全部的值）、xpath逻辑运算符定位
+clear()
+sleep(2)
+driver.find_element_by_xpath("//div[@class='item item-fore1' ]/input").send_keys("绝对路径钾元素class属性")
+sleep(2)
+driver.find_element_by_xpath("//div[@id='entry' ]/input").send_keys("绝对路径钾元素id属性")
+sleep(2)
+# 8>css定位
+
+
+# 等3秒
+sleep(3)
+# 关闭浏览器
+driver.quit()
\ No newline at end of file
Index: test_baidu.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test_baidu.py	(date 1549181942000)
+++ test_baidu.py	(date 1549181942000)
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# @Time    : 2019-02-03 15:38
+# @Author  : kuxui
+# @Site    : 
+# @File    : test_baidu.py
+# @Software: PyCharm
+
+
+from selenium import webdriver
+from time import sleep
+
+
+# 启动Chrome浏览器
+driver = webdriver.Chrome()
+
+# 打开百度页面
+driver.get("https://www.baidu.com/")
+
+# 等待3秒
+sleep(3)
+
+# 关闭浏览器
+driver.quit()
\ No newline at end of file
Index: Day5/read_file.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Day5/read_file.py	(date 1551940691000)
+++ Day5/read_file.py	(date 1551940691000)
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# @Time    : 2019-03-07 14:16
+# @Author  : Aries
+# @Site    : 
+# @File    : read_file.py
+# @Software: PyCharm
+#用于读取文件，text，Excel，xml
+#xlrd模块读取excel文件，xlwt模块用来写Excel（两个模块属于第三方模块需要安装）
+
+import xlrd
+
+class readfile():
+
+
+    #读取text
+    def readtext(path):
+        data = []
+        with open(path,'r') as f:
+            print("open file sucess !!!")
+            line = f.readline()
+            while line:
+                data.append(line.strip("\n"))
+                f.readline()
+        return data
+
+    #读取Excel
+    def readexcel(path):
+        data = []
+        workbook = xlrd.open_workbook(path)
+        #通过名字获取sheet页
+        sheet = workbook.sheet_by_name("")
+        nrows = sheet.nrows #行
+        ncols = sheet.ncols #列
+        for i in range(nrows):
+            data.append(sheet.cell_value(i,0))
+        return data
+
+
Index: Day5/develop_model.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Day5/develop_model.py	(date 1551939481000)
+++ Day5/develop_model.py	(date 1551939481000)
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# @Time    : 2019-03-06 10:29
+# @Author  : Aries
+# @Site    : 
+# @File    : develop_model.py
+# @Software: PyCharm
+#对网页的一些操作函数，登录，查询，校对
+
+
+from selenium import webdriver
+from time import sleep
+from selenium.common.exceptions import TimeoutException
+
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+from selenium.webdriver.common.by import By
+
+
+class model():
+
+
+    #登录
+    def login(driver,name,pwd):
+        try:
+            ele_loginname = WebDriverWait(driver,10,1).until(EC.presence_of_element_located((By.ID,"")))
+            ele_loginpwd = WebDriverWait(driver,10,1).until(EC.presence_of_element_located((By.ID, "")))
+            ele_login = WebDriverWait(driver, 10, 1).until(EC.presence_of_element_located((By.ID, "")))
+            ele_loginname.send_keys(name)
+            ele_loginpwd.send_keys(pwd)
+            sleep(3)
+            ele_login.click()
+            sleep(5)
+        except TimeoutException as e:
+            print(" login failled !!!")
+            print(e)
+
+    #搜索
+    def search(driver,key):
+        try:
+            ele_search_input = WebDriverWait(driver,10,1).until(EC.presence_of_element_located((By.ID,"")))
+            ele_search_input.send_key(key)
+            ele_search = WebDriverWait(driver, 10, 1).until(EC.presence_of_element_located((By.ID, "")))
+            ele_search.click()
+        except TimeoutException as e:
+            print("search failled !!!")
+            print(e)
+
+     #校验
+    def reasult_assert(driver,expected,actual):
+        try:
+            ele_actual = WebDriverWait(driver,10,1).until(EC.presence_of_element_located((By.ID,actual)))
+            if ele_actual == expected:
+                print("搜索成功")
+                return True
+            else:
+                print("搜索失败！！！")
+                return False
+
+        except TimeoutException as e:
+            print("search failled !!!")
+            print(e)
+
+
+
+
Index: jd_html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jd_html	(date 1550124582000)
+++ jd_html	(date 1550124582000)
@@ -0,0 +1,28 @@
+<input id="loginname" type="text" class="itxt" name="loginname"
+tabindex="1" autocomplete="off" value="" placeholder="邮箱/用户名/已验证手机">
+
+<input type="password" id="nloginpwd" name="nloginpwd"
+class="itxt itxt-error" tabindex="2" autocomplete="off" placeholder="密码">
+
+tag name定位
+tag name定位到元素为list，想取到制定的元素要知道元素是第几个，通过页面源码查找第几个
+
+linke text定位
+没有办法定位到input标签元素，linke text定位是定位a标签和文本元素
+
+xpath定位  绝对路径加元素属性定位（属性的值一定要写全部的值）
+driver.find_element_by_xpath("//div[@class='item item-fore1' ]/input").send_keys("绝对路径钾元素class属性")
+
+<div class="item item-fore1">
+     <label for="loginname" class="login-label name-label"></label>
+     <input id="loginname" type="text" class="itxt" name="loginname" tabindex="1" autocomplete="off" value="" placeholder="邮箱/用户名/已验证手机">
+     <span class="clear-btn"></span>
+</div>
+
+xpath逻辑运算符定位
+1.text()函数文本定位：【百度搜索按钮】
+        find_element_by_xpath("//*[text()='百度搜索']")
+2.and逻辑运算：[百度新闻链接]
+        find_element_by_xpath("//a[text()='新闻' and @class='mnav']")
+3.contains模糊定位：【百度新闻链接】
+        find_element_by_xpath("//*[contains(@href,'news.baidu0') and contains(@class,'mnav')]")
\ No newline at end of file
